Persistant-Storage-Snapshot/Snapshotter.java

```diff
@@ -0,0 +1,59 @@
+// add package here
+// Where should this file and StoredData be saved?
+//  In the same folder as Model.java?
+
+import codeu.chat.server.Model;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+
+// should I edit ServerMain for the optional flag part?
+//  edited with an if statement checking if StoredData.txt is empty?
+
+public class Snapshotter implements Runnable { 
+    // how should I go about creating and serializing Model.java
+
+    public void run() {
+        sleep();
```

Comment:
You want to sleep in the while the loop, this would just sleep once and never run again.